File: quest_back/src/main/java/quest/model/Matiere.java
package quest.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="matiere")
public class Matiere {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	@Column(length = 50,nullable = false)
	private String libelle;
	
	
	public Matiere() {}	
	
	public Matiere(Integer id, String libelle) {
		this.id = id;
		this.libelle = libelle;
	}

	public Matiere(String libelle) {
		this.libelle = libelle;
	}
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getLibelle() {
		return libelle;
	}

	public void setLibelle(String libelle) {
		this.libelle = libelle;
	}



	public String toString() {
		return "Matiere [id=" + id + ", libelle=" + libelle + "]";
	}
		
}

File: ANGULAR/demo-angular/src/app/matiere.ts
export class Matiere {
    constructor(private _id: number, private _libelle: string) { }

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get libelle(): string {
        return this._libelle;
    }

    public set libelle(value: string) {
        this._libelle = value;
    }
}
